x/posts/client/cli/tx.go

```diff
@@ -67,11 +80,38 @@ func GetCmdCreatePost(cdc *codec.Codec) *cobra.Command {
 				return err
 			}
 
-			msg := types.NewMsgCreatePost(args[1], parentID, allowsComments, args[0], map[string]string{}, from, time.Now().UTC())
+			msg := types.NewMsgCreatePost(
+				args[1],
+				parentID,
+				allowsComments,
+				args[0],
+				map[string]string{},
+				from,
+				time.Now().UTC(),
+				nil,
+			)
+
+			// If there are some medias
+			if len(args) > 3 {
+				var medias types.PostMedias
+				// Read each media and add it to the medias if valid
+				for i := 3; i < len(args); i++ {
```

Comment:
Sorry, but I've seen now that this won't work. If we have more than one media, this will be executed multiple times and only the latest media will be used. 

It should be changed to: 

```golang
if len(args) > 3 {
	medias := types.PostMedias{}

	// Read each media and add it to the medias if valid
	for i := 3; i < len(args); i++ {
		arg := strings.Split(args[i], ",")
		if len(arg) != 2 {
			return fmt.Errorf("if medias are specified, they shouldn't have empty fields, please use the --help flag to know more")
		}

		media := types.NewPostMedia(arg[0], arg[1])
		medias = medias.AppendIfMissing(media)
	}

	msg.Medias = medias
}
```

We need to initialize the `medias` to avoid a nil pointer error at line 105, also we can delete the second returned parameter from `AppendIfMissing` as it isn't necessary anymore.